utils/databake/src/primitives.rs

```diff
@@ -35,6 +35,17 @@ where
     T: Bake,
 {
     fn bake(&self, ctx: &CrateEnv) -> TokenStream {
+        if core::mem::size_of::<Self>() == core::mem::size_of::<&[u8]>()
+            && core::any::type_name::<Self>() == core::any::type_name::<&[u8]>()
+        {
+            // Safety: `&T` and `&[u8]` have the same size. Note that `self: &&T`, so this copies
```

Comment:
Thought: maybe we don't want to make `T: 'static`, but if we were okay doing that, then it's much easier to prove the safety by checking the TypeId instead of the type_name string

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=631461e873d8f78742dc20656a7301de

Reply:
We want to be able to bake things like `Foo<'data>`